pyad/reverse_mode.py

```diff
@@ -0,0 +1,167 @@
+import numpy as np
+
+class Reverse:
+
+	def __init__(self, value):
+		self.value = value
+		self.children = []
+		self.grad_value = None
+
+	def get_grad(self):
+		if self.grad_value is None:
+		    self.grad_value = sum(weight * var.get_grad() for weight, var in self.children)
+		return self.grad_value
+
+	def __neg__(self):
+		return self.__mul__(-1)
+
+	def __add__(self, other):
+		if isinstance(other, Reverse):
```

Comment:
Instead of having to do this `isinstance` check in every function, you could just convert the `other` to a `rev.Tensor` if it isn't one.

So, this `__add__` code might look like:
```
def __add__(self, other):
    other = other if isinstance(other, Tensor) else Tensor(other)
    res = Tensor(self.value + other.value)
    self.children.append((1, res))
    other.children.append((1, res))
    return res
```

Could you do something like that for `__add__` and the other functions in the file?

Reply:
Oops sorry I missed this one. I'll work on this later today.